### DISCUSSION TITLE
Nuxt 3 + NX. Modules

### DISCUSSION THREAD

**Discussion by alphaCastor** [nuxt/nuxt#16175] [Questions]
Hello. I am trying to set up monorepo and faced with some problems.
What I am trying to achieve is feature based structure with help of modules.

Basic structure is:

monorepo
- dist
- apps
- - nuxt-app
- libs
- - nuxt-module
- … rest of nx files like tsconfig.base.json, package.json, etc.

1st problem.
In NX monorepo all built apps and libs suppose to be in dist folder. Modifying nuxt.config.js allowed me to change apps output dir.
With modules is a bit harder. When I am trying to build or pack them using nuxt module builder it builds it in nuxt-module folder. Can I somehow define output directory for built module?

2nd problem
When lib is created, paths in tsconfig.base.json are updated (path to access this module).
But when I am trying to use this alias inside nuxt.config.js it says that this alias is not found.
For now there is workaround: I can define full path to module.ts file

3rd problem
Inside nuxt-module I have composables, components, pages. If I use path to src/module.ts all works fine. All this stuff will be registered in nuxt-app. But when I am trying to use path from dist/module.mjs it tells that there are problems with resolving those elements. So here it is a bit hard for me to understand what is going on and how I can deal with this.

And one more question.
For example I have 2 modules: module-a and module-b. Can I somehow use components from module-a in module-b?
I checked source codes of nuxt/kit and found some resolve utils where I can pass extensions. Is it related with what I am trying to achieve?

**Comment by gustawdaniel** [comment]
There is opened issue

https://github.com/ZachJW34/nx-plus/issues/236

I invested a lot of time in learning nx, and now I am disappointed, with how constrained it is.

**Reply by wobsoriano** [reply]
pnpm workspace <3

**Reply by alphaCastor** [reply]
> pnpm workspace <3

Do you any examples of how nuxt3 can be used with pnpm?

**Reply by wobsoriano** [reply]
@alphaCastor https://github.com/wobsoriano/nuxt-proxy

The important file there is `.npmrc` to make it work with Nuxt 3

**Reply by alphaCastor** [reply]
@wobsoriano Tried with another monorepo tool - Turborepo. It works with pnpm workspaces. And I managed to do all this staff with your config line. 
Thank you :)

**Reply by JFGHT** [reply]
> @wobsoriano Tried with another monorepo tool - Turborepo. It works with pnpm workspaces. And I managed to do all this staff with your config line. 
> 
> Thank you :)

Could you post a minimal repo with your solution?

**Reply by theolavaux** [reply]
I also tried Nx and I've been disappointed by the customization possible that indeed makes it hard to integrate with Nuxt 3. However if you're still looking into integrating this monorepo, check out this starter that uses Turborepo: [here](https://github.com/gurvan-guss/turborepo-nuxt-boilerplate)
I'm still getting is setup but it looks way more promising than Nx!
